source/common/http/http2/codec_impl_legacy.cc

```diff
@@ -46,6 +46,10 @@ class Http2ResponseCodeDetailValues {
   const absl::string_view inbound_empty_frame_flood = "http2.inbound_empty_frames_flood";
   // Envoy was configured to drop requests with header keys beginning with underscores.
   const absl::string_view invalid_underscore = "http2.unexpected_underscore";
+  // The upstream refused the stream.
+  const absl::string_view remote_refused = "http2.remote_refuse";
```

Comment:
As an aside, do you think we should be documenting these reason codes anywhere? It seems like it would be good to actually document these things in the user documentation somewhere (TODO/follow up fine).